x-pack/plugins/security_solution/public/common/components/toolbar/bulk_actions/use_bulk_alert_assignees_items.tsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { union } from 'lodash';
+import React, { useCallback, useMemo } from 'react';
+
+import { EuiFlexGroup, EuiIconTip, EuiFlexItem } from '@elastic/eui';
+import { ALERT_WORKFLOW_ASSIGNEE_IDS } from '@kbn/rule-data-utils';
+import type {
+  BulkActionsConfig,
+  RenderContentPanelProps,
+} from '@kbn/triggers-actions-ui-plugin/public/types';
+
+import { useAlertsPrivileges } from '../../../../detections/containers/detection_engine/alerts/use_alerts_privileges';
+import { ASSIGNEES_PANEL_WIDTH } from '../../assignees/constants';
+import { BulkAlertAssigneesPanel } from './alert_bulk_assignees';
+import * as i18n from './translations';
+import { useSetAlertAssignees } from './use_set_alert_assignees';
+
+export interface UseBulkAlertAssigneesItemsProps {
+  onAssigneesUpdate?: () => void;
+}
+
+export interface UseBulkAlertAssigneesPanel {
+  id: number;
+  title: JSX.Element;
+  'data-test-subj': string;
+  renderContent: (props: RenderContentPanelProps) => JSX.Element;
+  width?: number;
+}
+
+export const useBulkAlertAssigneesItems = ({
+  onAssigneesUpdate,
+}: UseBulkAlertAssigneesItemsProps) => {
+  const { hasIndexWrite } = useAlertsPrivileges();
+  const setAlertAssignees = useSetAlertAssignees();
+
+  const handleOnAlertAssigneesSubmit = useCallback(
+    async (assignees, ids, onSuccess, setIsLoading) => {
+      if (setAlertAssignees) {
+        await setAlertAssignees(assignees, ids, onSuccess, setIsLoading);
+      }
+    },
+    [setAlertAssignees]
+  );
+
+  const onSuccess = useCallback(() => {
+    onAssigneesUpdate?.();
+  }, [onAssigneesUpdate]);
+
+  const onActionClick = useCallback<Required<BulkActionsConfig>['onClick']>(
+    async (items, isSelectAllChecked, setAlertLoading, clearSelection, refresh) => {
```

Comment:
I think some of these params aren't used, like `clearSelection` and `refresh`.

Reply:
done